# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  cf2cef113c987419929e832aee7490e8841aada2
Feature: Multi-line: Allow forcing all folders off.

diff --git a/toolkit/content/widgets/tree.js b/toolkit/content/widgets/tree.js
--- a/toolkit/content/widgets/tree.js
+++ b/toolkit/content/widgets/tree.js
@@ -276,18 +276,19 @@
             currElement.getAttribute("label");
           popupChild.setAttribute("label", columnName);
           popupChild.setAttribute("colindex", currCol.index);
           if (currElement.getAttribute("hidden") != "true") {
             popupChild.setAttribute("checked", "true");
           }
           // Multi-line hack, this should go into the API.
           if (currCol.id == "subjectCol") {
-            let isMultiline = Services.prefs.getBoolPref("mail.pane_config.multiline_all", false);
-            if (isMultiline) {
+            let isMultilineAll = Services.prefs.getBoolPref("mail.pane_config.multiline_all", false);
+            let isMultilineNone = Services.prefs.getBoolPref("mail.pane_config.multiline_none", false);
+            if (isMultilineAll || isMultilineNone) {
               popupChild.setAttribute("disabled", "true");
             } else {
               popupChild.removeAttribute("disabled");
             }
           }
           // The subject can always be disabled to trigger multi-line view.
           if (currCol.primary && currCol.id != "subjectCol") {
             popupChild.setAttribute("disabled", "true");
